app/public/model/filtering/FilterModel.js

```diff
@@ -0,0 +1,121 @@
+            case 'exclude': {
+                this._activeFilters[field][type].push(value);
+                break;
+            }
+            case 'from':
+            case 'to': {
+                this._activeFilters[field][type] = [];
+                this._activeFilters[field][type].push(value);
+                break;
+            }
+            default: break;
+        }
+    }
+
+    resetFilterValue(field, type) {
+        this._activeFilters[field][type] = [];
+    }
+
+    prepareObjectStructure(field, type) {
+        if (! Object.prototype.hasOwnProperty.call(this._activeFilters, field)) {
+            this._activeFilters[field] = {
+                [type]: [],
+            };
+        }
+        if (! Object.prototype.hasOwnProperty.call(this._activeFilters[field], type)) {
+            this.resetFilterValue(field, type);
+        }
+    }
+
+    addFilter(field, value, type) {
+        this.prepareObjectStructure(field, type);
+        if (['from', 'to'].includes(type)) {
+            this.resetFilterValue(field, type);
+        }
+        this._activeFilters[field][type].push(value);
+        this.notify();
+    }
+
+    removeFilter(field, value, type) {
+        this._activeFilters[field][type] = this._activeFilters[field][type].filter((element) => element !== value);
+        this.notify();
+    }
+
+    filterTypesMapping(filterType, value) {
+        switch (filterType) {
+            case 'match': return `[or][like]=%${value}%`;
+            case 'exclude': return `[and][notLike]=%${value}%`;
+            case 'from': return `[gte]=${value}`;
+            case 'to': return `[lte]=${value}`;
+            default: return `=${value}`;
+        }
+    }
+
+    get activeFilters() {
+        return this._activeFilters;
+    }
+
+    get filterObjects() {
+        const result = Object.keys(this._activeFilters).map((field) =>
+            Object.keys(this._activeFilters[field]).reduce((typeAcc, currentType) => {
+                this._activeFilters[field][currentType].forEach((value) => typeAcc.push({
+                    field: field,
+                    type: currentType,
+                    value: value,
+                }));
+                return typeAcc;
+            }, []));
+        return result.flat();
+    }
+
+    isAnyFilterActive() {
+        return Object.keys(this._activeFilters).length > 0;
+    }
+
+    addSingleFilterPhrase(filterPhrase, field, type, value) {
+        return `${filterPhrase}${filterPhrase.length ? '&' : ''}filter[${field}]${this.filterTypesMapping(type, value)}`;
+    }
+
+    buildFilterPhrase() {
+        let filterPhrase = '';
+        for (const [field, typeValues] of Object.entries(this._activeFilters)) {
+            for (const [type, values] of Object.entries(typeValues)) {
+                filterPhrase = values.reduce((acc, value) => this.addSingleFilterPhrase(acc, field, type, value), filterPhrase);
+            }
+        }
+        return filterPhrase;
+    }
```

Comment:
I think that this way would be a bit more elegant:
```
buildSingleFilterPhrase(field, type, value) {
    return `$filter[${field}]${this.filterTypesMapping(type, value)}`;
}

buildFilterPhrase() {
    return this.filterObjects()
        .map(({ field, type, value }) => this.buildFilterPhrase(field, type, value))
        .join('&');
}
```

Reply:
No, it doesn't work this way. But I'll think of a prettier solution.

Reply:
Not exactly in a way you wrote, but I addressed the requested changes.